src/neo/Consensus/ConsensusContext.cs

```diff
@@ -107,12 +117,20 @@ public void Deserialize(BinaryReader reader)
                 Block.NextConsensus = null;
             Block.ConsensusData = reader.ReadSerializable<ConsensusData>();
             ViewNumber = reader.ReadByte();
+
             TransactionHashes = reader.ReadSerializableArray<UInt256>();
             Transaction[] transactions = reader.ReadSerializableArray<Transaction>(Block.MaxTransactionsPerBlock);
+
             PreparationPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
             CommitPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
             ChangeViewPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
             LastChangeViewPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
+
+            FuturePreparationPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
+            FutureCommitPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
+            FutureChangeViewPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
+            FutureRecoveryPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
```

Comment:
Payloads from different `BlockIndex` share the same arrays?

Reply:
Yes, we just store from the last known `BlockIndex` from that validator.
There is an attack from the Validator, however, quite easy to detect. Thus, we think it is a simple and good way to do that.

Reply:
I think we should allow multiple height from the same validator. We don't need so many arrays. We can use only one array to store all of them.

Reply:
O.o, we do not fork, @erikzhang, that does not make sense. aehuaheuaea

Why would we use any payload from a height lower than the last message of that `Validator`?

Reply:
In fact, @erikzhang, I think that you are right, maybe, in some peculiar cases, these can speed up the syncing.

Reply:
If we change to a single array, what is the best way to track positions?
Every time should we check all items of the array and delete those older and being used at current block index?

Reply:
I also think this is not very helpful, just as I think that saving future payloads is not very helpful. Because if we are missing some blocks, they should be solved by the block synchronization mechanism instead of "replaying" the consensus process.

Reply:
They can be ordered by the block index.

Reply:
@erikzhang, it is little bit different what you are proposing.
Use the "replaying" for block synchronization was not the purpose of this PR and that is why I did not implement this feature you are mentioning.

The propose here is mostly for a CN to contribute to the current block as soon as possible, not to use the information to create a block.

Reply:
In this sense, since you do not think it is helpful, neither me, better to leave the PR as it is right now.

In the way it is implemented it accomplishes the design it was made for, save promising payloads for when the node is synced up-to-date.